### DISCUSSION TITLE
Density --VS2017 Error

### DISCUSSION THREAD

**Discussion by yunhyejoung** [ImageMagick/ImageMagick#3368] [Development]
VS2017 (C++) Compile Error

I get an error when compiling Density("300") in VS2017 C++.

void CMFCApplication2Dlg::OnBnClickedButton1()
{
	const char *path = "C:\\Program Files\\ImageMagick-7.0.10-Q16-HDRI";
	InitializeMagick(path);
	Image worker;
	std::string src = "C:\\Users\\abc\\Pictures\\oh.bmp";
	std::string dest = "C:\\Users\\abc\\Pictures\\out.bmp";
	worker.read(dest);
	worker.resolutionUnits(PixelsPerInchResolution); //<======================   Error
	worker.density("300");
	worker.write(dest);
}

![error](https://user-images.githubusercontent.com/23716146/110234714-57703180-7f6f-11eb-8cb1-fd8aac0f0ff4.JPG)

![error1](https://user-images.githubusercontent.com/23716146/110234394-943b2900-7f6d-11eb-8596-c67bc77330e5.JPG)

![error2](https://user-images.githubusercontent.com/23716146/110234398-98ffdd00-7f6d-11eb-8f59-bcce007720a6.JPG)

![error3](https://user-images.githubusercontent.com/23716146/110234401-9d2bfa80-7f6d-11eb-97ae-b49827125668.JPG)

![error4](https://user-images.githubusercontent.com/23716146/110234475-090e6300-7f6e-11eb-838b-f2b6f0c8f66c.JPG)

![error5](https://user-images.githubusercontent.com/23716146/110234501-322ef380-7f6e-11eb-9125-e8e98556030c.JPG)

**Comment by dlemstra** [comment]
This doesn't look like a compiler error but a runtime error in your code? Looks like you should read from `src` instead of `dest`?

**Reply by yunhyejoung** [reply]
In Compile VS 2017 

void CMFCApplication2Dlg::OnBnClickedButton1()
{
const char *path = "C:\Program Files\ImageMagick-7.0.10-Q16-HDRI";
InitializeMagick(path);
Image worker;
std::string src = "C:\Users\abc\Pictures\oh.bmp";
std::string dest = "C:\Users\abc\Pictures\out.bmp";
worker.read(src);
worker.resolutionUnits(PixelsPerInchResolution); //<====================== Error
worker.density("300");
worker.write(dest);
}

**Reply by dlemstra** [reply]
It looks like something else is happening according to your screenshot. I suspect that you failed to read the image?

**Reply by yunhyejoung** [reply]
I read the image well.

It works Code fine in VS 2010.

But, In VS 2017 , an error occurs as shown in the picture above.
